src/gauging.jl

```diff
@@ -73,114 +78,58 @@ function vidal_gauge(
   return ψ_vidal, bond_tensors
 end
 
-"""Use an ITensorNetwork ψ in the symmetric gauge and its mts to put ψ into the vidal gauge. Return the bond tensors and ψ_vidal."""
 function vidal_gauge(
   ψ::ITensorNetwork,
-  pψψ::PartitionedGraph,
-  mts;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  edges=NamedGraphs.edges(ψ),
-  svd_kwargs...,
+  bond_tensors;
+  (cache!)=nothing,
+  cache_update_kwargs=default_cache_update_kwargs(cache!),
+  kwargs...,
 )
-  bond_tensors = initialize_bond_tensors(ψ)
-  return vidal_gauge(
-    ψ,
-    pψψ,
-    mts,
-    bond_tensors;
-    eigen_message_tensor_cutoff,
-    regularization,
-    edges,
-    svd_kwargs...,
-  )
-end
+  if isnothing(cache!)
+    cache! = Ref(default_cache(ψ))
+  end
 
-"""Put an ITensorNetwork into the vidal gauge (by computing the message tensors), return the network and the bond tensors. Will also return the mts that were constructed"""
-function vidal_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  verbose=false,
-  svd_kwargs...,
-)
-  ψψ = norm_network(ψ)
-  pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
+  cache![] = update(cache![]; cache_update_kwargs...)
 
-  mts = belief_propagation(pψψ; niters, target_precision=target_canonicalness, verbose)
-  return vidal_gauge(
-    ψ, pψψ, mts; eigen_message_tensor_cutoff, regularization, svd_kwargs...
-  )
+  return vidal_gauge(ψ, bond_tensors, cache![]; kwargs...)
+end
+
+"""Put an ITensorNetwork into the vidal gauge (by computing the message tensors), return the network and the bond tensors."""
+function vidal_gauge(ψ::ITensorNetwork; kwargs...)
+  bond_tensors = initialize_bond_tensors(ψ)
+  return vidal_gauge(ψ, bond_tensors; kwargs...)
 end
 
 """Transform from an ITensor in the Vidal Gauge (bond tensors) to the Symmetric Gauge (partitionedgraph, message tensors)"""
-function vidal_to_symmetric_gauge(ψ::ITensorNetwork, bond_tensors::DataGraph)
+function symmetric_gauge(ψ::ITensorNetwork, bond_tensors::DataGraph)
```

Comment:
I think we should design this function to allow passing a `cache!` keyword argument, as opposed to building it internally and then returning it, to make the interface consistent across the the gauging functions. Ideally all of the gauging functions would be designed that way, and in fact I would imagine many `ITensorNetwork` functions could use that same design, such as the solver code, code for computing expectation values, etc.

Reply:
Sorry, scratch that. I think in this case it makes sense to just pass in the Vidal gauge bond tensors, and then output the BP cache for the symmetric gauge, since in some sense the Vidal gauge bond tensors are playing the role of the cache. This points towards needing a better system for dealing with the Vidal gauge (i.e. a devoted Vidal gauge tensor network type), but that's a separate issue.

Reply:
I'm going back and forth about this issue, mostly because of my initial issue with this code, which was that it's not always clear which functions are potentially updating the cache or improving the gauging, vs. just converting naively from one gauge to another without performing any BP updates. In other words, I find that it is unclear and hard to keep track of when a function assumes it is in the idealized version of a gauge (i.e. assumes it is in a proper Vidal gauge or symmetric gauge, with a small gauge error), vs. when it improves the messages or gauge while doing the conversion. I think we should clarify that point and come up with a design which makes that choice clear, either through function names or keyword arguments, which of those choices is being made when you call one of these gauging functions.